controller/array_action/array_mediator_ds8k.py

```diff
@@ -595,11 +592,15 @@ def get_object_by_id(self, object_id, object_type):
             return self._generate_snapshot_response_with_verification(api_object)
         return self._generate_volume_response(api_object)
 
-    def create_snapshot(self, name, volume_name, pool_id=None):
-        logger.info("creating snapshot '{0}' from volume '{1}'".format(name, volume_name))
-        target_api_volume = self._create_snapshot(name, pool_id, source_volume_name=volume_name)
-        logger.info("finished creating snapshot '{0}' from volume '{1}'".format(name, volume_name))
-        return self._generate_snapshot_response(target_api_volume, volume_name)
+    @convert_scsi_id_to_array_id
+    def create_snapshot(self, volume_id, snapshot_name, pool):
+        logger.info("creating snapshot '{0}' from volume '{1}'".format(snapshot_name, volume_id))
+        source_volume = self._get_api_volume_by_id(volume_id)
```

Comment:
consider renaming `source_volume` to `source_api_volume`

Reply:
done